packages/blade/src/components/Carousel/__tests__/Carousel.web.test.tsx

```diff
@@ -440,4 +441,22 @@ describe('Carousel Snapshots', () => {
 
     expect(container).toMatchSnapshot();
   });
+
+  it('should not render overlay on mobile devices', () => {
+    const viewport = mockViewport({ width: '320px', height: '568px' });
+
+    const { container } = renderWithTheme(
+      <Carousel scrollOverlayColor="surface.background.level1.lowContrast" shouldAddStartEndSpacing>
+        <CarouselItem>
+          <TestimonialCard />
+        </CarouselItem>
+        <CarouselItem>
+          <TestimonialCard />
+        </CarouselItem>
+      </Carousel>,
+    );
+
+    expect(container).toMatchSnapshot();
```

Comment:
match snapshot won't catch it na. what we need is to detect if the overlay was actually rendered or not? probably some data attribute or something since its a cosmetic thing?

Reply:
It will catch. Because we won't even render the CSS :before wala part if `showOverlay` is false. 

<img width="748" alt="image" src="https://github.com/razorpay/blade/assets/35374649/99c4e3c2-d4f2-4c4f-b682-7312e17d430e">

Reply:
no i mean once someone changes anything it'll be hard to detect from snapshot if that will be visual breakage, no?